Content.Shared/Weapons/Reflect/ReflectSystem.cs

```diff
@@ -200,4 +203,30 @@ private void OnReflectHandUnequipped(Entity<ReflectComponent> ent, ref GotUnequi
         ent.Comp.InRightPlace = false;
         Dirty(ent);
     }
+
+    #region Examine
+    private void OnExamine(Entity<ReflectComponent> ent, ref ExaminedEvent args)
+    {
+        // This isn't examine verb or something just because it looks too much bad.
+        // Trust me, universal verb for the potential weapons, armor and walls looks awful.
+        var value = MathF.Round(ent.Comp.ReflectProb * 100, 1);
+
+        if (!_toggle.IsActivated(ent.Owner) || value == 0 || ent.Comp.Reflects == ReflectType.None)
+            return;
+
+        var compTypes = ent.Comp.Reflects.ToString().Split(", ");
+
+        List<string> typeList = [];
```

Comment:
A micro-optimization - initialize the List with the needed capacity since we know it in advance:
```suggestion
        List<string> typeList = new(compTypes.Length);
```